### PR TITLE
test(liquidate): dont realize bad debt

### PR DESCRIPTION
**Pull request by Rubilmax**
it was a simple one but I ended up refactoring it to avoid stack too deep and harmonize both tests

### CODE REVIEW

File: test/forge/integration/LiquidateIntegrationTest.sol [hunk 1]
```diff
@@ -66,134 +66,130 @@ contract LiquidateIntegrationTest is BaseTest {
...
-        vm.assume(amountCollateral > 1);
-
-        amountSupplied = bound(amountSupplied, amountBorrowed, MAX_TEST_AMOUNT);
-        _supply(amountSupplied);
+    function testLiquidateSharesInputNoBadDebt(LiquidateTestParams memory params, uint256 sharesRepaid) public {
+        _setLltv(_boundTestLltv(params.lltv));
+        (params.amountCollateral, params.amountBorrowed, params.priceCollateral) =
+            _boundUnhealthyPosition(params.amountCollateral, params.amountBorrowed, params.priceCollateral);
 
-        uint256 expectedBorrowShares = amountBorrowed.toSharesUp(0, 0);
```
Comment (MerlinEgalite) [inline test/forge/integration/LiquidateIntegrationTest.sol:151]: was it this line the issue?

**Review by MerlinEgalite — COMMENTED** [review]
It's hard to tell what is related to the refactor and what is related to the fix with all the changes

File: test/forge/integration/LiquidateIntegrationTest.sol [hunk 2]
```diff
@@ -66,134 +66,130 @@ contract LiquidateIntegrationTest is BaseTest {
...
+        morpho.supplyCollateral(marketParams, params.amountCollateral, BORROWER, hex"");
+        morpho.borrow(marketParams, params.amountBorrowed, 0, BORROWER, BORROWER);
         vm.stopPrank();
 
-        oracle.setPrice(priceCollateral);
+        oracle.setPrice(params.priceCollateral);
+
+        uint256 borrowShares = morpho.borrowShares(id, BORROWER);
+        uint256 liquidationIncentiveFactor = _liquidationIncentiveFactor(marketParams.lltv);
+        uint256 maxSharesRepaid = (params.amountCollateral - 1).mulDivDown(params.priceCollateral, ORACLE_PRICE_SCALE)
```
Comment (Rubilmax) [inline test/forge/integration/LiquidateIntegrationTest.sol:161]: this was the fix: not realize bad debt

Comment (MerlinEgalite) [reply test/forge/integration/LiquidateIntegrationTest.sol:161]: the minus 1?

**Review by Jean-Grimal — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

File: test/forge/integration/LiquidateIntegrationTest.sol [hunk 3]
```diff
@@ -66,134 +66,130 @@ contract LiquidateIntegrationTest is BaseTest {
...
+    struct LiquidateTestParams {
+        uint256 amountCollateral;
+        uint256 amountSupplied;
+        uint256 amountBorrowed;
+        uint256 priceCollateral;
+        uint256 lltv;
+    }
 
-        vm.assume(amountCollateral > 1);
+    function testLiquidateSeizedInputNoBadDebt(LiquidateTestParams memory params, uint256 amountSeized) public {
```
Comment (MathisGD) [inline test/forge/integration/LiquidateIntegrationTest.sol:77]: the name could be clearer. Are we testing a liquidation when the account isn't in a bad debt situation or when the liquidator doesn't realize the bad debt?

File: test/forge/integration/LiquidateIntegrationTest.sol [hunk 4]
```diff
@@ -66,134 +66,130 @@ contract LiquidateIntegrationTest is BaseTest {
...
-    ) public {
-        _setLltv(_boundTestLltv(lltv));
-        (amountCollateral, amountBorrowed, priceCollateral) =
-            _boundUnhealthyPosition(amountCollateral, amountBorrowed, priceCollateral);
-
-        vm.assume(amountCollateral > 1);
-
-        amountSupplied = bound(amountSupplied, amountBorrowed, MAX_TEST_AMOUNT);
-        _supply(amountSupplied);
+    function testLiquidateSharesInputNoBadDebt(LiquidateTestParams memory params, uint256 sharesRepaid) public {
```
Comment (MathisGD) [inline test/forge/integration/LiquidateIntegrationTest.sol:138]: same remark

File: test/forge/integration/LiquidateIntegrationTest.sol (hunk 3 above)
Comment (MathisGD) [reply test/forge/integration/LiquidateIntegrationTest.sol:77]:
```suggestion
    function testLiquidateSeizedInputNoBadDebtRealization(LiquidateTestParams memory params, uint256 amountSeized) public {
```

File: test/forge/integration/LiquidateIntegrationTest.sol (hunk 4 above)
Comment (MathisGD) [reply test/forge/integration/LiquidateIntegrationTest.sol:138]:
```suggestion
    function testLiquidateSharesInputNoBadDebtRealization(LiquidateTestParams memory params, uint256 sharesRepaid) public {
```

**Review by MathisGD — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

**Pull request merged**